deps/v8/src/builtins/builtins-microtask-queue-gen.cc

```diff
@@ -198,11 +198,20 @@ void MicrotaskQueueBuiltinsAssembler::RunSingleMicrotask(
     const TNode<Object> thenable = LoadObjectField(
         microtask, PromiseResolveThenableJobTask::kThenableOffset);
 
+    // Run the promise before/debug hook if enabled.
+    RunPromiseHook(Runtime::kPromiseHookBefore, microtask_context,
```

Comment:
I'm looking through the history of promise hooks (v8:4643 and such) trying to understand why this would've been left out and I haven't found anything, but it does seem suspicious...

Reply:
It looks like it was just missed. I think "thenables" just weren't considered. I don't see anything in  v8:4643 suggesting otherwise.

Reply:
Agreed. They have it for the other branches in there, but not thenables. Makes me think there could be a specific _reason_ for that. As far as I can tell though it's just been overlooked. In testing it locally though, it fixes the problem perfectly for _our_ needs.

Reply:
I think there was a discussion at diagnostics summit in munich if v8 should call promise hooks for non user visible promises or not. The agreement was it is not needed as it improves performance and assumption was that no hooks are needed for "invisible" promises. Promisehooks are quite expensive for v8 - in special since they did more optimizations of promises/await that time.
To me this looks like some corner case regarding visibility of the promise created here. The Promise itself is not visible in the sense that user can access it as it is only used within the `await` statemachine.  On the other hand the side effects are visible.